uber/models.py

```diff
@@ -439,14 +439,18 @@ def order(self, attrs):
                 order.append(col.desc() if attr.startswith('-') else col)
             return self.order_by(*order)
 
-        def icontains(self, attr=None, val=None, **filters):
-            query = self
+        def icontains_predicate(self, attr=None, val=None, **filters):
```

Comment:
Correct, they're intended to be exactly equivalent. However, `search()` now calls directly into icontains_predicate, because it needs a sort of unique type of filter that wasn't possible with the old icontains method:

    filter((search[0] like first_name && search[1] like last_name) || (search[0]+search[1] like legal_name))

more specifically, there's this wacky "or" in there - so I extracted the logic for creating the expression into its own method, and left the old icontains with the same behavior for all the other use sites, then for search I did:

    filter(icontains_pred(first, last) || icontains_pred(legal))

the code of which is on line 738ish in the diff.

I agree that docs for behavior would be good. As a side note, I feel like if we're modeling `icontains` off of the SQLAlchemy `contains` method, we definitely shouldn't be doing anything with `filter` - it should do what the `icontains_predicate` is doing, and returning a boolean expression instead of filtering. That can be dealt with later, though.